Dockerfile

```diff
@@ -1,22 +1,44 @@
-FROM python:2.7-slim
+FROM alpine:3.3
+MAINTAINER Wonderfall <wonderfall@opmbx.org>
 
-WORKDIR /app
+ENV BASE_URL=False IMAGE_PROXY=False
 
-RUN useradd searx
+COPY . /usr/local/searx
 
-EXPOSE 5000
-CMD ["/usr/local/bin/uwsgi", "--uid", "searx", "--gid", "searx", "--http", ":5000", "-w",  "searx.webapp"]
+RUN apk -U add \
+    build-base \
+    python \
+    python-dev \
+    py-pip \
+    libxml2 \
+    libxml2-dev \
+    libxslt \
+    libxslt-dev \
+    libffi-dev \
+    openssl \
+    openssl-dev \
+    ca-certificates \
+ && adduser -D -h /usr/local/searx -s /bin/sh searx searx \
+ && cd /usr/local/searx \
+ && pip install --no-cache -r requirements.txt \
+ && sed -i "s/127.0.0.1/0.0.0.0/g" searx/settings.yml \
+ && sed -i "s/ultrasecretkey/`openssl rand -hex 16`/g" searx/settings.yml \
+ && sed -i "s|base_url : False|base_url : $BASE_URL|g" searx/settings.yml \
+ && sed -i "s/image_proxy : False/image_proxy : $IMAGE_PROXY/g" searx/settings.yml \
+ && apk del \
+    build-base \
+    python-dev \
+    py-pip\
+    libffi-dev \
+    openssl-dev \
+    libxslt-dev \
+    libxml2-dev \
+    openssl-dev \
+    ca-certificates \
+ && chown -R searx:searx /usr/local/searx \
+ && rm -f /var/cache/apk/*
```

Comment:
I cannot really see why having less layers is "always preferred". IMO this is less readable and it loses when it comes to image rebuilding. 

adduser, pip, sed and chown commands have nothing to do with the  apk command, are small and fast commands and can be separated into multiple layers.

Also if you break them into multiple cmds you can move COPY after the package install and therefore reuse the package install layers on new searx versions